### PR TITLE
feat(watch): `--include` & `--exclude` CLI flags

### LINKED ISSUE #181: Add files to be watched
### Feature request

I would like to be able to add files to be watched in addition to the default, for example to watch the `.env` file.

### Why?

I use a `.env` file with `dotenv` for environment variables. Often I'll change a value in `.env` during runtime to test something (e.g. enable/disable feature flag). 

With nodemon I'm using:

```
nodemon --watch src --watch .env src/index.ts
```

It would be nice to be able to do this with tsx as well.

### Alternatives

My current alternative is nodemon+ts-node+swc. But tsx seems like a more convenient all-in-one solution.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by boxizen**
closes #181 

This change adds additional files to the watchlist via adding the `add` parameter.

### CODE REVIEW

File: tests/specs/watch.ts [hunk 1]
```diff
@@ -300,5 +300,59 @@ export default testSuite(async ({ describe }, { tsx }: NodeApis) => {
...
 			}, 10_000);
 		});
+
+		describe('include', ({ test }) => {
+			test('watch files for changes', async () => {
+				const entryFile = 'index.js';
+				const file = 'file';
+				const value = 'test';
+				const append = `${value}append`;
+				const fixture = await createFixture({
```
Comment (privatenumber) [inline tests/specs/watch.ts:310]: Use the `await using` syntax so you can remove the `fixture.rm()` below

File: tests/specs/watch.ts [hunk 2]
```diff
@@ -300,5 +300,59 @@ export default testSuite(async ({ describe }, { tsx }: NodeApis) => {
...
+					}
+				});
+
+				const tsxProcessResolved = await tsxProcess;
+				await fixture.rm();
+
+				expect(tsxProcessResolved.stdout).not.toMatch(`current date: ${value}`);
+				expect(tsxProcessResolved.stderr).toBe('');
+			}, 10_000);
+		});
```
Comment (privatenumber) [inline tests/specs/watch.ts:356]: Can you add a test for `exclude`?

File: tests/specs/watch.ts [hunk 3]
```diff
@@ -300,5 +300,59 @@ export default testSuite(async ({ describe }, { tsx }: NodeApis) => {
...
+						import fs from 'fs';
+						fs.readFile('./${file}', (err, data) => {
+							if (!err) {
+								console.log(data.toString());
+							} else {
+								console.log(error);
+							}
+						});
+					`.trim(),
+					[file]: `${value}`,
```
Comment (privatenumber) [inline tests/specs/watch.ts:321]:
```suggestion
					[file]: value,
```

File: tests/specs/watch.ts [hunk 4]
```diff
@@ -300,5 +300,59 @@ export default testSuite(async ({ describe }, { tsx }: NodeApis) => {
...
+
+		describe('include', ({ test }) => {
+			test('watch files for changes', async () => {
+				const entryFile = 'index.js';
+				const file = 'file';
+				const value = 'test';
+				const append = `${value}append`;
+				const fixture = await createFixture({
+					[entryFile]: `
+						import fs from 'fs';
```
Comment (privatenumber) [inline tests/specs/watch.ts:312]:
```suggestion
						import fs from 'fs/promises';
```

File: tests/specs/watch.ts [hunk 5]
```diff
@@ -300,5 +300,59 @@ export default testSuite(async ({ describe }, { tsx }: NodeApis) => {
 				expect(p.stderr).toBe('');
 			}, 10_000);
 		});
+
+		describe('include', ({ test }) => {
+			test('watch files for changes', async () => {
+				const entryFile = 'index.js';
+				const file = 'file';
+				const value = 'test';
+				const append = `${value}append`;
```
Comment (privatenumber) [inline tests/specs/watch.ts:309]: I see you're trying to couple these together via variables but I think the test will be much more readable if you just hard-code them

**Comment by mdmower-csnw** [comment]
@boxizen @privatenumber - I'm interested in these changes. Is there any way I could help here? I cannot push directly to this branch since it is on boxizen's repo, but I could create a new branch that continues from this one if boxizen doesn't have time. The code review comments all look easy enough to address (also, TIL that [TS5.2 already introduced support](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-2.html) for [`using` declarations](https://github.com/tc39/proposal-explicit-resource-management))! Thanks and let me know how I could help.

**Comment by privatenumber** [comment]
Closing as it was implemented via https://github.com/privatenumber/tsx/pull/625

**Pull request closed without merging**